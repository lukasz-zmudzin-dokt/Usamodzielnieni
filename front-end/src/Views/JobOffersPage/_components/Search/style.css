@media (min-width: 768px) {
  .search__wrapper {
    display: flex;
    justify-content: space-around;
  }
}
