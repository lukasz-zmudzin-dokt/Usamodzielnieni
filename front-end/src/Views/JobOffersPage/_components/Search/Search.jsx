import React, { useState } from "react";
import { Form, Button, Row } from "react-bootstrap";
import "./style.css";

const voivodeships = [
  "Zachodnio-pomorskie",
  "Pomorskie",
  "Warmińsko-Mazurskie",
  "Podlaskie",
  "Lubuskie",
  "Wielkopolskie",
  "Kujawsko-pomorskie",
  "Mazowieckie",
  "Lubelskie",
  "Dolnośląskie",
  "Łódzkie",
  "Opolskie",
  "Świętokrzystkie",
  "Śląskie",
  "Małopolskie",
  "Podkarpackie"
];

const typeOfWork = ["Szkolenie", "Praca", "Staż"];
const earningsTab = ["Płatne", "Bezpłatne"];

const Search = ({ offers, setFilter }) => {
  const [voivodeship, setVoivodeship] = useState("-- Wybierz --");
  const [companyName, setCompanyName] = useState("-- Wybierz --");
  const [type, setType] = useState("-- Wybierz --");
  const [earnings, setEarnings] = useState("-- Wybierz --");

  const filter = event => {
    event.preventDefault();
    const filters = { voivodeship, companyName, type, earnings };
    let newTab = offers;
    let check = 0;
    for (let key in filters) {
      if (filters[key] !== "-- Wybierz --") {
        newTab = newTab.filter(val => {
          return val[key] === filters[key];
        });
      } else check++;
    }
    if (check != 4) setFilter({ tab: newTab, changed: true });
  };

  const deleteFilter = e => {
    setVoivodeship("-- Wybierz --");
    setCompanyName("-- Wybierz --");
    setType("-- Wybierz --");
    setEarnings("-- Wybierz --");
    setFilter(offers);
  };
  let companies = [...new Set(offers.map(({ companyName }) => companyName))];
  return (
    <Form className="mb-2" onSubmit={filter}>
      <div className="search__wrapper">
        <Form.Group controlId="exampleForm.ControlSelect2">
          <Form.Label>Województwo</Form.Label>
          <Form.Control
            as="select"
            onChange={e => setVoivodeship(e.target.value)}
            value={voivodeship}
          >
            <option data-hidden="true" disabled>
              -- Wybierz --
            </option>
            {voivodeships.map(val => (
              <option key={val}>{val}</option>
            ))}
          </Form.Control>
        </Form.Group>
        <Form.Group controlId="exampleForm.ControlSelect2">
          <Form.Label>Firma</Form.Label>
          <Form.Control
            as="select"
            onChange={e => setCompanyName(e.target.value)}
            value={companyName}
          >
            <option data-hidden="true" disabled>
              -- Wybierz --
            </option>
            {companies.map(val => (
              <option key={val}>{val}</option>
            ))}
          </Form.Control>
        </Form.Group>
        <Form.Group controlId="exampleForm.ControlSelect2">
          <Form.Label>Szkolenie/Praca</Form.Label>
          <Form.Control
            as="select"
            onChange={e => setType(e.target.value)}
            value={type}
          >
            <option data-hidden="true" disabled>
              -- Wybierz --
            </option>
            {typeOfWork.map(val => (
              <option key={val}>{val}</option>
            ))}
          </Form.Control>
        </Form.Group>
        <Form.Group controlId="exampleForm.ControlSelect2">
          <Form.Label>Płatne/Bezpłatne</Form.Label>
          <Form.Control
            as="select"
            onChange={e => setEarnings(e.target.value)}
            value={earnings}
          >
            <option data-hidden="true" disabled>
              -- Wybierz --
            </option>
            {earningsTab.map(val => (
              <option key={val}>{val}</option>
            ))}
          </Form.Control>
        </Form.Group>
      </div>
      <Row className="justify-content-center align-items-center m-0">
        <Button
          variant="secondary"
          type="submit"
          className="pl-4 pr-4 pt-2 pb-2 mr-5"
        >
          Filtruj
        </Button>
        <Button
          variant="secondary"
          onClick={deleteFilter}
          className="pl-4 pr-4 pt-2 pb-2"
        >
          Usuń filtr
        </Button>
      </Row>
    </Form>
  );
};

export default Search;
