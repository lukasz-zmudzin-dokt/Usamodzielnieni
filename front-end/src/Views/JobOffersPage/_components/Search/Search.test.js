import React from "react";
import { render } from "@testing-library/react";
import Search from "Views/JobOffersPage/_components/Search";
import renderer from "react-test-renderer";

const offers = [
  {
    id: "1",
    title: "Stolarz poszukiwany!",
    description:
      "Do naszego zakładu potrzebujemy osoby, która ma chęć rąbać drewno! To możesz być ty!!!",
    companyName: "Rębacze z Cintry sp. z o.o.",
    firstName: "Jarosław",
    lastName: "Psikuta",
    email: "paniewidzisztamsnakońcu@gmail.com",
    phone: "133792137",
    voivodeship: "Lubelskie",
    type: "Szkolenie",
    earnings: "Bezpłatne"
  },
  {
    id: "2",
    title: "Pszczelarz poszukiwany!",
    description:
      "Do naszego zakładu potrzebujemy osoby, która ma chęć wyciagac miody pszczolom! To możesz być ty!!!",
    companyName: "Miody sp. z o.o.",
    firstName: "Jarosław",
    lastName: "Psikuta",
    email: "paniewidzisztamsnakońcu@gmail.com",
    phone: "133792137",
    voivodeship: "Lubelskie",
    type: "Praca",
    earnings: "Płatne"
  }
];

describe("Search", () => {
  it("should render without crashing", () => {
    render(<Search offers={offers} setFilter={() => null} />);
  });
});

it("renders correctly", () => {
  const search = renderer
    .create(<Search offers={offers} setFilter={() => null} />)
    .toJSON();
  expect(search).toMatchSnapshot();
});
