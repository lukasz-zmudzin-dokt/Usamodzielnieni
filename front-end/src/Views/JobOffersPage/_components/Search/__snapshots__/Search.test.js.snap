// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form works(search) 1`] = `null`;

exports[`renders correctly 1`] = `
<form
  className="mb-2"
  onSubmit={[Function]}
>
  <div
    className="search__wrapper"
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlSelect2"
      >
        Województwo
      </label>
      <select
        className="form-control"
        id="exampleForm.ControlSelect2"
        onChange={[Function]}
        value="-- Wybierz --"
      >
        <option
          data-hidden="true"
          disabled={true}
        >
          -- Wybierz --
        </option>
        <option>
          Zachodnio-pomorskie
        </option>
        <option>
          Pomorskie
        </option>
        <option>
          Warmińsko-Mazurskie
        </option>
        <option>
          Podlaskie
        </option>
        <option>
          Lubuskie
        </option>
        <option>
          Wielkopolskie
        </option>
        <option>
          Kujawsko-pomorskie
        </option>
        <option>
          Mazowieckie
        </option>
        <option>
          Lubelskie
        </option>
        <option>
          Dolnośląskie
        </option>
        <option>
          Łódzkie
        </option>
        <option>
          Opolskie
        </option>
        <option>
          Świętokrzystkie
        </option>
        <option>
          Śląskie
        </option>
        <option>
          Małopolskie
        </option>
        <option>
          Podkarpackie
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlSelect2"
      >
        Firma
      </label>
      <select
        className="form-control"
        id="exampleForm.ControlSelect2"
        onChange={[Function]}
        value="-- Wybierz --"
      >
        <option
          data-hidden="true"
          disabled={true}
        >
          -- Wybierz --
        </option>
        <option>
          Rębacze z Cintry sp. z o.o.
        </option>
        <option>
          Miody sp. z o.o.
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlSelect2"
      >
        Szkolenie/Praca
      </label>
      <select
        className="form-control"
        id="exampleForm.ControlSelect2"
        onChange={[Function]}
        value="-- Wybierz --"
      >
        <option
          data-hidden="true"
          disabled={true}
        >
          -- Wybierz --
        </option>
        <option>
          Szkolenie
        </option>
        <option>
          Praca
        </option>
        <option>
          Staż
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="exampleForm.ControlSelect2"
      >
        Płatne/Bezpłatne
      </label>
      <select
        className="form-control"
        id="exampleForm.ControlSelect2"
        onChange={[Function]}
        value="-- Wybierz --"
      >
        <option
          data-hidden="true"
          disabled={true}
        >
          -- Wybierz --
        </option>
        <option>
          Płatne
        </option>
        <option>
          Bezpłatne
        </option>
      </select>
    </div>
  </div>
  <div
    className="justify-content-center align-items-center m-0 row"
  >
    <button
      className="pl-4 pr-4 pt-2 pb-2 mr-5 btn btn-secondary"
      disabled={false}
      type="submit"
    >
      Filtruj
    </button>
    <button
      className="pl-4 pr-4 pt-2 pb-2 btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Usuń filtr
    </button>
  </div>
</form>
`;
